cli/generator-op-microservice/generators/microservice/index.js

```diff
@@ -0,0 +1,326 @@
+'use strict';
+const Generator = require('yeoman-generator');
+const chalk = require('chalk');
+const yosay = require('yosay');
+const fs = require('fs');
+const _ = require('lodash');
+const mkdirp = require('mkdirp');
+const packageJson = require('../../package.json');
+
+let graphqlSupport;
+let dbSupport;
+
+module.exports = class extends Generator {
+  prompting() {
+    // Have Yeoman greet the user.
+    this.log(
+      yosay(`Welcome to the One Platform Microservice generator ${chalk.red('generator-op-microservice')}!`)
+    );
+
+    const prompts = [
+      {
+        type: 'input',
+        name: 'name',
+        message: 'Your microservice name'
+      },
+      {
+        type: 'list',
+        name: 'type',
+        message: 'Which type of API do you want to create?',
+        choices: [
+          {
+            name: 'GraphQL',
+            value: 'graphql'
+          },
+          {
+            name: 'REST',
+            value: 'rest'
+          }
+        ],
+        default: this.options.type,
+        when: () => !this.options.type
+      },
+      {
+        type: 'confirm',
+        name: 'dbSupport',
+        message: 'Are you going to use MongoDB?',
+        default: true
+      },
+      {
+        type: 'input',
+        name: 'authorName',
+        message: 'Author name'
+      },
+      {
+        type: 'input',
+        name: 'authorEmail',
+        message: 'Author Email'
+      }
```

Comment:
Can we make it use the `git config user.username` and `git config user.email` as default values? If nothin is provided?

Reply:
I like the idea of using git config to fetch authorship details. @riginoommen can you implement it here?

Reply:
#55 - Opened issue regarding this.